### ISSUE TITLE
liveBroadcast wont work

### ISSUE THREAD

**Issue by ataogluburak** [mgp25/Instagram-API#1729]
[log - Kopya.txt](https://github.com/mgp25/Instagram-API/files/1539677/log.-.Kopya.txt)
[log.txt](https://github.com/mgp25/Instagram-API/files/1539678/log.txt)

**Comment by ataogluburak** [comment]
and my function;

	function liveBroadcast($filename){
		global $username, $password;

		/////// CONFIG ///////
		$debug = false;
		$truncatedDebug = false;
		//////////////////////

		/////// MEDIA ////////
		$videoFilename = "./videos/".$filename;
		//////////////////////

		$ig = new \InstagramAPI\Instagram($debug, $truncatedDebug);

		try {
		    $ig->login($username, $password);
		} catch (\Exception $e) {
		    echo 'Something went wrong: '.$e->getMessage()."\n";
		    exit(0);
		}

		try {

		    $stream = $ig->live->create();
		    $ig->live->start($stream->getBroadcastId());

		    exec(
		        'ffmpeg -rtbufsize 256M -re -i '
		        .escapeshellarg($videoFilename)
		        .' -acodec libmp3lame -ar 44100 -b:a 128k -pix_fmt yuv420p -profile:v baseline -s 720x1280 -bufsize 6000k -vb 400k -maxrate 1500k -deinterlace -vcodec libx264 -preset veryfast -g 30 -r 30 -f flv "rtmp://live-upload.instagram.com:80/rtmp'
		        .substr($stream->getUploadUrl(), 42)
		        .'"'
		    );

		    $ig->live->end($stream->getBroadcastId());
		    $ig->live->addToPostLive($stream->getBroadcastId());

		} catch (\Exception $e) {
		    echo 'Something went wrong: '.$e->getMessage()."\n";
		}
	}

**Comment by SteveJobzniak** [comment]
Ok when reading your log, there are no Instagram errors. So I think you simply do not have ffmpeg.

**Comment by SteveJobzniak** [comment]
Hey. I needed to do something at the computer today, so I took the chance to rewrite the livestream example. Now the code is safe and does error checking.

https://github.com/mgp25/Instagram-API/commit/e8101b1d8929d571b265079cc34a139ae6ef5b68

Just download this new source code:

https://github.com/mgp25/Instagram-API/blob/master/examples/liveBroadcast.php

Whatever is wrong with your ffmpeg command (most likely it's missing) will be detected automatically now with this new code.

Take care!
